### DISCUSSION TITLE
Dockerfile sets a different userid

### DISCUSSION THREAD

**Discussion by jeremybanzhaf** [renovatebot/renovate#32338] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab, Latest Renovate

### Please tell us more about your question or problem

[a6d93dc](https://github.com/renovatebot/renovate/commit/a6d93dcbc61a0d08149c4034d896d3efdb00be06) sets `USER` to `12021` instead of `1000`.  This causes permissions issues when the renovate container is run with a persistent volume.  Is there a need for 12021?  Thanks in advance!

**Comment by jeremybanzhaf** [comment]
Related: https://github.com/renovatebot/renovate/discussions/32317

**Comment by rarkins** [comment, marked as answer]
Yes, we have chosen to use a high UID by default, to reduce chances of conflict. You may _change_ that UID at runtime if you wish, using normal container UID concepts, and it should still work.

Is there some restriction which says "USER 1000 may run with persistent volumes but other users may not"? Which specific restriction are you running into?

**Comment by jeremybanzhaf** [comment]
Aha, thanks.  There are of course no restrictions on userids, but when there's an existing persistent volume with directories owned by user 1000, 12021 does not have permission to the existing cache.

**Reply by rarkins** [reply]
Are you able to chown or delete it this one time?

**Reply by jeremybanzhaf** [reply]
We'll have to delete it. Thanks for the context!
